lib/game.rb

```diff
@@ -6,27 +6,50 @@
 require_relative "./game_checker"
 
 class Game
-  attr_reader :display, :board, :player_1
-  def initialize(display, presenter, board, players)
-    @display = display
-    @presenter = presenter
-    @board = board
-    @player_1 = players[0]
-    @player_2 = players[1]
+  attr_reader :display, :board, :player_1, :player_2, :active_player
+  def initialize(config_object)
+    @display = config_object[:display]
+    @presenter = config_object[:presenter]
+    @board = config_object[:board]
+    @player_1 = config_object[:player_1]
+    @player_2 = config_object[:player_2]
+    @game_checker = config_object[:game_checker]
 
     @active_player = @player_1
+    @game_over = false
   end
 
   def start
     @display.output(WELCOME)
+    print_board
   end
 
   def play
-    print_board
-    @board.place_mark(@player_1.mark, @player_1.get_input(@display))
-    print_board
-    @board.place_mark(@player_2.mark, @player_2.get_input(@display))
-    print_board
+    until @game_over
+      @board.place_mark(@active_player.mark, @active_player.get_input(@display))
+      print_board
+      if @game_checker.game_over?(@board, @player_1, @player_2)
+        @game_over = true
+        game_over
+      else
+        switch_player
+      end
+    end
+  end
+
+  def switch_player
+    @active_player = @active_player == @player_1 ? @player_2 : @player_1
+  end
+
+  def game_over
+    @display.output(GAME_OVER)
+    if @game_checker.tie?(@board, @player_1, @player_2)
+      @display.output(TIE)
+    elsif @game_checker.winner?(@board, @player_1)
+      @display.output(PLAYER1_WINS)
+    else
+      @display.output(PLAYER2_WINS)
+    end
```

Comment:
Do you think displaying the outcome belongs in game, game_checker or presenter?

Reply:
Kind of thought it belonged here? This is like the Welcome message. Display takes in the Presenter as an argument in `print_board`, do you mean I should make Presenter have a `game_end` or something? Like instead of just displaying the constants, having it go `@display.output(@presenter.end_game)`?